Add Minecraft 1.21.11 items and update dependencies

RenaudRl opened the pull request:
Added new Minecraft 1.21.11 items including spears, nautilus armors, netherite horse armor, and new spawn eggs with corresponding assets and model definitions. Updated Paper API, Netty SocketIO, and PacketEvents dependencies to support 1.21.11 and Netty 4.2. Adjusted socket authentication to use AuthorizationResult and disabled websocket compression to resolve Netty version conflicts.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added 1.21.11 items: seven spear variants, five nautilus armor variants, netherite horse armor, and four spawn eggs.

* **Chores**
  * Platform and build updates for Minecraft 1.21.11; enabled parallel/configure-on-demand Gradle builds; dependency bumps.

* **Refactor**
  * Ticker wiring switched to a descriptive mode notifier for improved mute/enable handling.

* **Chores**
  * Websocket compression disabled in communication configuration.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

RenaudRl committed:
Add Minecraft 1.21.11 items and update dependencies

Added new Minecraft 1.21.11 items including spears, nautilus armors, netherite horse armor, and new spawn eggs with corresponding assets and model definitions. Updated Paper API, Netty SocketIO, and PacketEvents dependencies to support 1.21.11 and Netty 4.2. Adjusted socket authentication to use AuthorizationResult and disabled websocket compression to resolve Netty version conflicts.

RenaudRl committed:
Merge branch 'develop' into 1.21.11

gabber235 commented on engine/engine-paper/build.gradle.kts:
```diff
@@ -30,12 +30,12 @@ repositories {
     api(project(":engine-core"))
     api(project(":engine-loader"))
 
-    compileOnlyApi("com.corundumstudio.socketio:netty-socketio:1.7.19") // Keep this on a lower version as the newer version breaks the ping
```
This cannot change. It will break the entire webpanel communication if you do this.

RenaudRl replied:
Oh dear, my mistake! I ran several tests but couldn't connect to the panel on 1.21.11 without updating Netty.

[00:57:24 WARN]: [io.netty.channel.DefaultChannelPipeline] An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.lang.NoSuchMethodError: 'io.netty.handler.codec.compression.ZlibDecoder io.netty.handler.codec.compression.ZlibCodecFactory.newZlibDecoder(io.netty.handler.codec.compression.ZlibWrapper, int)'
        at io.netty.handler.codec.http.websocketx.extensions.compression.DeflateDecoder.decompressContent(DeflateDecoder.java:117) ~[netty-codec-http-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.handler.codec.http.websocketx.extensions.compression.DeflateDecoder.decode(DeflateDecoder.java:83) ~[netty-codec-http-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.handler.codec.http.websocketx.extensions.compression.PerMessageDeflateDecoder.decode(PerMessageDeflateDecoder.java:91) ~[netty-codec-http-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.handler.codec.http.websocketx.extensions.compression.PerMessageDeflateDecoder.decode(PerMessageDeflateDecoder.java:31) ~[netty-codec-http-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:91) ~[netty-codec-4.1.118.Final.jar:4.1.118.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:356) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at com.corundumstudio.socketio.transport.PollingTransport.channelRead(PollingTransport.java:109) ~[?:?]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:356) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at com.corundumstudio.socketio.handler.AuthorizeHandler.channelRead(AuthorizeHandler.java:137) ~[?:?]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:356) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:102) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:356) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346) ~[netty-codec-4.1.118.Final.jar:4.1.118.Final]
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318) ~[netty-codec-4.1.118.Final.jar:4.1.118.Final]
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:356) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1429) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:918) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:168) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.handle(AbstractNioChannel.java:445) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.channel.nio.NioIoHandler$DefaultNioRegistration.handle(NioIoHandler.java:388) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.channel.nio.NioIoHandler.processSelectedKey(NioIoHandler.java:596) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.channel.nio.NioIoHandler.processSelectedKeysOptimized(NioIoHandler.java:571) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.channel.nio.NioIoHandler.processSelectedKeys(NioIoHandler.java:512) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.channel.nio.NioIoHandler.run(NioIoHandler.java:484) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.channel.SingleThreadIoEventLoop.runIo(SingleThreadIoEventLoop.java:225) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.channel.SingleThreadIoEventLoop.run(SingleThreadIoEventLoop.java:196) ~[netty-transport-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:1193) ~[netty-common-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.2.7.Final.jar:4.2.7.Final]
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) ~[netty-common-4.2.7.Final.jar:4.2.7.Final]
        at java.base/java.lang.Thread.run(Thread.java:1570) ~[?:?]

gabber235 commented on engine/engine-paper/src/main/java/com/typewritermc/engine/paper/TypewriterPaperLoader.java:
```diff
@@ -18,32 +18,31 @@ public class TypewriterPaperLoader implements PluginLoader {
         addDependency(classpathBuilder, "org.jetbrains.kotlin:kotlin-stdlib:2.2.10", central);
         addDependency(classpathBuilder, "org.jetbrains.kotlin:kotlin-reflect:2.2.10", central);
         addDependency(classpathBuilder, "org.jetbrains.kotlinx:kotlinx-coroutines-core:1.10.2",
-            central);
-        addDependency(classpathBuilder, "com.corundumstudio.socketio:netty-socketio:1.7.19",
-            central);
```
You cannot change this as it will completely break the webpanel.

gabber235 reviewed (changes requested):
The old socketio had a good reason because the web panel side had issues that is why we kept it at a lower version. If it needs to change then I need extensive proof that it won't break the entire web panel. 

Something like 10+ hours of continues use and doing basically everything on it.

RenaudRl committed:
Downgrade netty-socketio and update ticker handling

Downgraded netty-socketio from 2.0.13 to 1.7.19 in build scripts and dependency loading to avoid ping issues and Netty API conflicts. Updated CommunicationHandler to use Boolean for authentication instead of AuthorizationResult, and improved ticker handling in ticker.dart to use TickerModeData and new notifier methods.

RenaudRl committed:
Merge branch '1.21.11' of https://github.com/RenaudRl/Typewriter into 1.21.11

RenaudRl committed:
Refactor _updateTickers method for null safety

Kerzinator24 committed:
Merge branch 'develop' into 1.21.11

Kerzinator24 committed:
rename item icons to match file list
